doc/smart-switch/reboot/reboot-hld.md

```diff
@@ -0,0 +1,337 @@
+
+* DPUs will send an acknowledgment to the NPU and then undergo a reboot. After receiving the acknowledgment from the DPUs, the NPU will proceed to reboot itself to complete the overall reboot procedure. The vendor-specific reboot API should include an error handling mechanism to manage DPU reboot failures. Additionally log all the failures. DPUs will be in DPU_READY state, if the reboot happened successfully.
+
+* Upon successful reboot, the NPU resumes operation. As part of the post-reboot process, the NPU may choose to rescan the PCI devices. This rescan operation,
+performed either by invoking vendor API or by echoing '1' to the /sys/bus/pci/devices/XXXX:XX:XX.X/rescan file, ensures that all PCI devices are properly
+recognized and initialized.
+
+## High-Level Design ##
+
+### ModuleBase Class API ###
+
+reboot(self, reboot_type):
+```
+Define new reboot_type as MODULE_REBOOT_DPU for DPU only reboot and MODULE_REBOOT_SMARTSWITCH for entire switch reboot.
+```
+
+This API is defined in [smartswitch-pmon.md](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/pmon/smartswitch-pmon.md#:~:text=reboot(self%2C%20reboot_type)%3A)
+
+### ModuleBase Class new APIs ###
+
+pci_detach_dpu(self):
+```
+Detaches the DPU PCI device from the NPU. In the case of non-smart-switch chassis, no action is taken.
+
+Returns:
+    True
+```
+
+pci_reattach_dpu(self):
+```
+Rescans the PCI bus and attach the DPU back to NPU. In the case of non-smart-switch chassis, no action is taken.
+
+Returns:
+    True
+```
+
+get_dpu_bus_info(self, dpu_module_name):
+```
+For a given DPU module name, retrieve the PCI bus information. In the case of non-smart-switch chassis, no action is taken.
+
+Returns:
+    Returns the PCI bus information in BDF format like "[DDDD:]:BB:SS.F"
+```
+
+### NPU platform.json ###
+
+Introduce a new parameter, <span style="color:blue">'dpu_halt_services_timeout'</span>, to specify the duration(in secs) for waiting for the DPU to terminate all services, as defined by
+the platform vendor. If the DPU fails to respond within this timeout, the NPU will proceed with the reboot sequence. If no timeout is explicitly
+defined, a default timeout will be used.
+
+```json
+{
+    .
+    .
+    "dpu_halt_services_timeout" : "TBD"
+
+    "DPUs" : [
+        {
+            "dpu0": {
+                "bus_info" : "[DDDD:]BB:SS.F"
+            }
+        },
+        {
+            "dpu1": {
+                "bus_info" : "[DDDD:]BB:SS.F"
+            }
+        },
+        .
+        .
+        {
+            "dpuX": {
+                "bus_info" : "[DDDD:]BB:SS.F"
+            }
+        }
+    ]
+    .
+    .
+}
+```
+
+### GNOI API implementation ###
+
+According to the RebootRequest protocol outlined below, we will utilize the HALT command in the RebootMethod to terminate services on the DPU.
+When the NPU sends the RebootRequest with the HALT RebootMethod to the DPU, it will kill all services except GNMI and database services.
```

Comment:
@vvolam how will the pre-shutdown be executed on the DPU? Will it be done via `/usr/local/bin/reboot` script? I think it necessary to use this script as it already knows which SONiC services to shut down and can also run the FW upgrade if needed.

Reply:
@oleksandrivantsiv  That's correct, reboot script only will take care of services shutdown.

Reply:
@vvolam as discussed in the meeting the `/usr/local/bin/reboot` script should be invoked on the DPU to shut down SONiC services and run FW upgrade if needed.